core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceUnitTests.scala

```diff
@@ -549,7 +560,416 @@ class WorkspaceServiceUnitTests extends AnyFlatSpec with OptionValues with Mocki
+        (SamWorkspaceRoles.writer, Set[SamResourceRole](), false),
+        (SamWorkspaceRoles.writer, Set[SamResourceRole](SamWorkspaceRoles.shareWriter), true)
+      )
+    ) { (highestAccessLevel: SamResourceRole, additionalRoles: Set[SamResourceRole], canShareResult: Boolean) =>
+      val workspaceSamResource = SamUserResource(
+        workspace.workspaceId,
+        SamRolesAndActions(Set(highestAccessLevel), Set()),
+        SamRolesAndActions(additionalRoles, Set()),
+        SamRolesAndActions(Set(), Set(SamWorkspaceActions.compute)),
+        Set(),
+        Set()
+      )
+      val sam = mock[SamDAO]
+      when(sam.listUserResources(SamResourceTypeNames.workspace, defaultRequestContext))
+        .thenReturn(Future(Seq(workspaceSamResource)))
+      val wsRepo = mock[WorkspaceRepository]
+      when(wsRepo.listWorkspacesByIds(ArgumentMatchers.eq(Seq(workspace.workspaceIdAsUUID)), any))
+        .thenReturn(Future(Seq(workspace)))
+      val wsm = mock[WorkspaceManagerDAO]
+      when(wsm.listWorkspaces(defaultRequestContext)).thenReturn(List())
+      val service = workspaceServiceConstructor(
+        samDAO = sam,
+        workspaceManagerDAO = wsm,
+        workspaceRepository = wsRepo
+      )(defaultRequestContext)
+
+      val params = WorkspaceFieldSpecs(fields = Some(Set("workspace", "public", "accessLevel", "canShare")))
+      val result = Await.result(service.listWorkspaces(params, -1), Duration.Inf)
+
+      val resultWorkspace = result match {
+        case jsa: JsArray =>
+          jsa.elements.headOption.map {
+            _.convertTo[WorkspaceListResponse]
+          }
+        case _ => None
+      }
+      resultWorkspace.get.canShare.get shouldBe canShareResult
+    }
+  }
+
+  it should "return the matching value of canCompute" in {
+    forAll(
+      Table(
+        ("cloudPlatform", "highestAccessLevel", "additionalRoles", "canComputeResult"),
+        (WorkspaceCloudPlatform.Azure, SamWorkspaceRoles.owner, Set[SamResourceRole](), true),
+        (WorkspaceCloudPlatform.Azure, SamWorkspaceRoles.projectOwner, Set[SamResourceRole](), true),
+        (WorkspaceCloudPlatform.Azure, SamWorkspaceRoles.writer, Set[SamResourceRole](), true),
+        // can compute is not valid for azure workspaces - the user needs at least the writter permission
+        (WorkspaceCloudPlatform.Azure,
+         SamWorkspaceRoles.reader,
+         Set[SamResourceRole](SamWorkspaceRoles.canCompute),
+         false
+        ),
+        (WorkspaceCloudPlatform.Gcp, SamWorkspaceRoles.owner, Set[SamResourceRole](), true),
+        (WorkspaceCloudPlatform.Gcp, SamWorkspaceRoles.projectOwner, Set[SamResourceRole](), true),
+        (WorkspaceCloudPlatform.Gcp, SamWorkspaceRoles.writer, Set(SamWorkspaceRoles.canCompute), true),
+        (WorkspaceCloudPlatform.Gcp, SamWorkspaceRoles.writer, Set[SamResourceRole](), false),
+        (WorkspaceCloudPlatform.Gcp, SamWorkspaceRoles.reader, Set(SamWorkspaceRoles.canCompute), true),
```

Comment:
Technically this situation shouldn't be allowed, although I know we just looked at how someone can get into this situation when a writer attempts to modify other users' access to a workspace. I'm not sure if it makes sense to include it as a test case if it's an unsupported situation

Reply:
I included it because being exhaustive on a table driven test is trivial, and the code supports the situation.